### ISSUE TITLE
feat(pipeline): arbitration layer — wire policy registry + reconcile + input-shape routing (pipeline ≥ v0 by construction)

### ISSUE THREAD

**Issue by GirlBossRush** [isp-nexus/mailwoman#478]
**The parity capstone — zero GPU.** Parity is a pipeline property, not a model property. The arena scorecard's `v0-only` column (14% clean / 8% noisy / 21% edge) is whole parses v0 gets that we drop, and the machinery to capture them exists but sits unwired. Wiring calibrated per-component arbitration makes pipeline parity ≥ v0 **by construction**.

## The four pieces (all exist, none finished)

1. **Policy registry** (`core/policy/`): defaults every component to `rule_only`; per-component promotion requires code changes. Add a config surface (per-locale per-tag mode: `rule_only | neural_only | rule_preferred | neural_preferred` + confidence threshold) loadable without code edits. **`applyPreferenceFilters` (~30 lines, the dedup core) has zero tests — test it first.**
2. **Reconcile (Stage 5)**: `ClassifierCandidate` top-k contract is still a mock (`core/pipeline/reconcile.ts`) — no production path emits it. Land the real classifier top-k emission (`parseWithLogits` exists; see parser-hardening issue for the repair-divergence fix that must land first).
3. **Input-shape routing**: the routing truth (clean→rules, noisy→neural, both-weak→abstain to resolver) has been in the docs since #15. `kind-classifier` + `query-shape` are the shipped input-shape signals; `#244` (coarse-placer/abstention) is the router's abstention tier. Wire them as the arbitration prior.
4. **Calibrated confidence as the arbiter**: isotonic calibration shipped (#59/#367) — use the calibrated posterior, not raw logits, as the cross-source comparison currency.

## Gate (pre-registered)

Re-run the 3 capability arenas with arbitration on: `v0-only` → ~0 in every arena (every v0 win captured), `neural-only` retained, `both-fail` unchanged or better. Per-locale F1 floors hold. Demo presets stable.

## Scope guard

No retrain. No new classifiers. This is wiring + tests + config surface for machinery that already exists. Context: `2026-06-10-DEEP-DIVE-REVIEW.md` §2, parity scorecard 2026-06-09 lens 1.

**Comment by GirlBossRush** [comment]
## Night-10 groundwork: design notes (the zero-GPU slice; implementation follows post-review)

**Fresh evidence this layer is the right next move (all measured tonight):**
- v4.2.0 arena `v0-only` cells: clean 17%, noisy 9%, edge 21% — every one a whole parse v0 gets that the pipeline currently throws away.
- NAD source-independent holdout: v0 locality-name 86.7 vs neural 77.3, while neural region 100.0 vs v0 91.7 — per-component complementarity in one row.
- The policy registry's preference machinery now has its sharp-case tests (#481 part 1: merged-source survival, threshold-before-preference ordering — both verified correct).

### 1. Policy config surface (design)

A `policies` block, loadable at pipeline construction (no code edits to roll out a tag):

```jsonc
// mailwoman.policies.json — per-locale per-tag; absent = today's defaults (rule_only)
{
  "en-US": {
    "region":   { "mode": "neural_preferred", "confidence_threshold": 0.6 },
    "locality": { "mode": "both" },               // arbitration decides downstream
    "street":   { "mode": "rule_preferred" }      // clean-input street stays rules until #492
  }
}
```

Loader: `InMemoryPolicyRegistry.fromConfig(json)` — validates tags against the ComponentTag union (unknown tag = loud failure), modes against the PolicyMode union, then `withDefaults().set(...)` per entry. `createRuntimePipeline({ policyConfig })` threads it. ~60 lines + tests. **Note from tonight's test-writing:** `set()` silently accepts unknown fields (my `minimumConfidence` typo cost 20 minutes) — `fromConfig` must reject unknown keys.

### 2. Input-shape routing memo

The routing truth (clean→rules, noisy→neural) becomes a PRIOR, not a hard route: kind-classifier + query-shape emit a shape signal; the arbitration maps shape → per-tag mode overlays (e.g. `shape=canonical_postal` overlays `street: rule_preferred`). Overlay composition: base policies ← shape overlay ← caller overrides, same merge semantics as locale profiles. The abstention tier (#244) slots as `shape=coarse_query` → skip fine tags entirely.

### 3. Target metric (pre-registered now, before implementation)

Re-run the three arenas with arbitration on: **`v0-only` → ~0 in every arena** (every v0 win captured), `neural-only` retained, per-locale floors hold, presets stable. The complementarity rows above are the existence proof; the metric is the contract.

### Sequencing

Config surface (1) is implementable immediately post-#481-part-1. The reconcile real-emission (the `parseWithLogits` consumer) now consumes repaired tokens identical to the user path — the divergence blocker is resolved. Routing overlay (2) after (1). Not attempted at end-of-shift; this comment is the reviewable artifact.

**Comment by GirlBossRush** [comment]
**Two amendments from this morning's measurements:** (1) Step 1 (config surface) is implemented — PR referencing this issue; unknown-field rejection included per the design note. (2) **The routing prior from design-note §2 is obsolete before implementation**: the gaz-fed arena re-measurement (parity-scorecard-2026-06-10, correction block) killed 'clean→rules' — neural now leads clean (41 vs 29) AND noisy (71 vs 39); v0 leads only rare edge formats (po-box/military/rural). The shape overlay therefore simplifies from a clean/noisy split to a single narrow rule: `shape ∈ {edge formats} → rule_preferred; else neural-leaning`. Less machinery than designed, and the v0-only bounty cells are now 9/5/16. Pre-registered target metric (§3) unchanged.

**Comment by GirlBossRush** [comment]
**Corrected implementation map (the deep-dive's 'mocked contract' claim was STALE — falsified by reading the live wiring):** `runtime-pipeline.ts` already emits real `ClassifierCandidate`s via `aggregateSpanLogits(parseWithLogits(...))` → `reconcileSpans`, DEFAULT-ON since Route A Phase II (2026-06-07, regate report). With #500's repairs-in-both, the tokens are user-path-consistent. What genuinely remains, re-scoped: (1) **the concordance axes are unwired** — the runtime call passes no `resolverCandidates`/`parentChain`; wiring Stage 6's backend into those two lookups activates the scoring the reconciler already implements. (2) **Rules aren't in the runtime path at all** — the arbitration vision lands not as a new layer but as a SECOND CANDIDATE EMITTER: rule classifiers → `ClassifierCandidate[]` (calibrated scores via the policy registry's thresholds), merged with the neural top-k; `reconcileSpans` already handles multi-candidate spans. Reconcile IS the arbitration engine. (3) The shape overlay stays as designed (one narrow edge-format rule post-arena-flip). Step 1 (config loader) merged; this re-scope makes step 2 smaller and step 3 (rules-as-emitter) the capstone. Target metric unchanged: arena v0-only → ~0.

**Comment by GirlBossRush** [comment]
Direction note added (operator instinct, 2026-06-11): **the sub-premise nexus** — PO Box / Apt / Suite / Unit / Level share one DESIGNATOR+IDENTIFIER grammar, now provenance-tracked across five codex locales. Three escalating homes, all pre-tagging-as-INFORMATION (the #464 override rejection stands): a subpremise clue dim on the gazetteer channel; a Stage 2.7 sub-premise proposer (sharing machinery with #518's bracket/paren revival — one proposer, two cue families); and this issue's capstone, where codex matchers become the first second-emitter candidate vocabulary with the proposer as dress rehearsal. `docs/articles/plan/2026-06-11-subpremise-proposer-direction.md`

**Comment by GirlBossRush** [comment]
Closing with receipts (operator-approved, 2026-07-02 triage): the arbitration promotion was **falsified by the coordinate gate** — #685 measured arena +122 label-wins against coord p50 3.3 → 1069 km and street+HN precondition 100 → 48%; the machinery stays merged default-OFF (#682–#684) and the runtime-flag register marks `arbitrate` + the policy-registry modes **research-only** (revisit only with a coordinate-gated design). The 'pipeline ≥ v0 by construction' goal was reached by a different route: argmax decode + the resolver's joint-consistency passes. The operator's sub-premise-nexus direction note (one DESIGNATOR+IDENTIFIER grammar across PO Box/Apt/Suite/Unit) moves to the #901 campaign as a candidate shard/proposer — commented there.
